Forceful deletion of Shoots (#8414)
gardener/gardener: docs/usage/shoot_operations.md

It will also be removed even if the restart of one or more services failed.

> ℹ️ In the example mentioned above, you could additionally verify when/whether the kubelet restarted by using `kubectl describe node <node-name>` and looking for such a `Starting kubelet` event.

> ## Shoot Force Deletion

Reviewer: ```suggestion
## Force Deletion
```

> When the `ShootForceDeletion` feature gate in the gardener-apiserver is enabled, users will be able to force-delete the Shoot. This is only possible if the Shoot fails to be deleted normally. For forceful deletion, the following conditions should be met:

Reviewer 2: ```suggestion
When the `ShootForceDeletion` feature gate in the gardener-apiserver is enabled, users will be able to force-delete the Shoot. This is only possible if the Shoot fails to be deleted normally. For forceful deletion, the following conditions must be met:
```

- Shoot has a deletion timestamp.
- Shoot status contains at least one of the following [ErrorCodes](shoot_status.md#error-codes):
  - ERR_CLEANUP_CLUSTER_RESOURCES
  - ERR_CONFIGURATION_PROBLEM
  - ERR_INFRA_DEPENDENCIES
  - ERR_INFRA_UNAUTHENTICATED

>   - ERR_INFRA_UNAUTHORIZED

Reviewer: You could `tick` them

If the above conditions are satisfied, you can annotate the Shoot with `confirmation.gardener.cloud/force-deletion=true`, and Gardener will cleanup the Shoot controlplane and the Shoot metadata.

> > :warning: You **MUST** ensure that all the resources created in the IaaS account are cleaned up to prevent orphaned resources. Gardener will **NOT** delete any resources in the Shoot cloud-provider account.

Reviewer: ```suggestion
> :warning: You **MUST** ensure that all the resources created in the IaaS account are cleaned up to prevent orphaned resources. Gardener will **NOT** delete any resources in the underlying infrastructure account. Hence, use this annotation on your own risk and only if you are fully aware of these consequences.
```

> > :warning: You **MUST** ensure that all the resources created in the IaaS account are cleaned up to prevent orphaned resources. Gardener will **NOT** delete any resources in the underlying infrastructure account. Hence, use this annotation on your own risk and only if you are fully aware of these consequences.

Reviewer 3: ```suggestion
> :warning: You **MUST** ensure that all the resources created in the IaaS account are cleaned up to prevent orphaned resources. Gardener will **NOT** delete any resources in the underlying infrastructure account. Hence, use this annotation at your own risk and only if you are fully aware of these consequences.
```
